src/mscorlib/src/System/Array.cs

```diff
@@ -1073,6 +1073,42 @@ public static T[] Empty<T>()
             return Array.FindIndex(array, match) != -1;
         }
 
+        public static void Fill<T>(T[] array, T value)
+        {
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            }
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = value;
+            }
+        }
+
+        public static void Fill<T>(T[] array, int startIndex, int count, T value)
```

Comment:
@stephentoub, good point. I just checked and the generic `IndexOf` accepts `(T[], T, int, int)`. It sounds reasonable to put the value first.